plasmapy/atomic/particle_input.py

```diff
@@ -218,10 +238,33 @@ def wrapper(*args, **kwargs):
 
             funcname = wrapped_function.__name__
 
-            args_to_become_particles = [
-                argname for argname in annotations.keys()
-                if annotations[argname] is Particle and argname != 'return'
-            ]
+            args_to_become_particles = []
+            for argname in annotations.keys():
+                if isinstance(annotations[argname], tuple):
+                    annotated_argnames = annotations[argname]
+                    expected_params = len(annotated_argnames)
+                    received_params = len(arguments[argname])
+                    if not expected_params == received_params:
+                        raise TypeError(
+                            f"Number of parameters allowed in the tuple "
+                            f"({expected_params} parameters) are "
+                            f"not equal to number of parameters passed in "
+                            f"the tuple ({received_params} parameters).")
+                elif isinstance(annotations[argname], list):
+                    annotated_argnames = annotations[argname]
+                    expected_params = len(annotated_argnames)
+                    if expected_params > 1:
+                        raise TypeError(
+                            f"A list can take any number of parameters "
```

Comment:
Maybe "Put in [Particle] as the annotation to accept an arbitrary number of Particle arguments."

I love the idea though

Reply:
Nice. That makes it easier to understand. I think maybe "arbitrary number of Particle elements" would be a better choice than "arbitrary number of Particle arguments". What do you think?

Reply:
Nah, I think we should rather keep the 'elements' key word for iron, hydrogen etc here :smiley:

Reply:
Haha, ok got it.